src/main/python/ttconv/imsc/elements.py

```diff
@@ -96,7 +96,7 @@ def is_instance(xml_elem) -> bool:
     raise NotImplementedError
 
   @staticmethod
-  def from_xml(parent_ctx, xml_elem):
+  def from_xml(parent_ctx, xml_elem, progress_callback = None):
```

Comment:
Any reason for not moving `progress_callback` into `parent_ctx`?

Reply:
I don't think putting this in the parent_ctx is the right thing to do. There isn't the same parallel for the writer (from_model) 

Also, the parent context is currently owned/created in class TTElement(TTMLElement): from_xml. 

The caller who pass in the progress bar is the tt.py layer which passes it into reader to_model which passes in a None for parent context via the call imsc_elements.TTElement.from_xml

Give that, making the parent context own it, seems odd to me.